### PR TITLE
[#1611] Add MissingElse cop

### PR DESCRIPTION
**Pull request by rrosenblum**

### CODE REVIEW

File: config/disabled.yml [hunk 1]
```diff
@@ -19,6 +19,15 @@ Style/MethodCalledOnDoEndBlock:
   StyleGuide: 'https://github.com/bbatsov/ruby-style-guide#single-line-blocks'
   Enabled: false
 
+Style/MissingElse:
+  Description: >-
+                Require if statements to have an else condition.
+                If enabled, it is recommended that
+                Style/UnlessElse and Style/EmptyElse be enabled.
+                This will conflict with Style/EmptyElse if
+                Style/EmptyElse is configured to both
```
Comment (bbatsov) [inline config/disabled.yml]: to style both (sounds clearer this way)

**Comment by bbatsov** [comment]
You'll need to rebase.

File: config/disabled.yml [hunk 2]
```diff
@@ -19,6 +19,15 @@ Style/MethodCalledOnDoEndBlock:
   StyleGuide: 'https://github.com/bbatsov/ruby-style-guide#single-line-blocks'
   Enabled: false
 
+Style/MissingElse:
+  Description: >-
+                Require if statements to have an else condition.
```
Comment (bbatsov) [inline config/disabled.yml]: if **expressions**; else **branch**
You should also mention `case`.

**Comment by bbatsov** [comment]
I wonder if there should be some way via configuration to make this process only `if` or `case` expressions.

**Comment by rrosenblum** [comment]
@bbatsov, The branch has been rebased, and updated to support configurable styles.

File: config/disabled.yml [hunk 3]
```diff
@@ -19,6 +19,23 @@ Style/MethodCalledOnDoEndBlock:
   StyleGuide: 'https://github.com/bbatsov/ruby-style-guide#single-line-blocks'
   Enabled: false
 
+Style/MissingElse:
+  Description: >-
+                Require if expressions to have an else branches.
```
Comment (bbatsov) [inline config/disabled.yml]: if/case

File: config/disabled.yml [hunk 4]
```diff
@@ -19,6 +19,23 @@ Style/MethodCalledOnDoEndBlock:
...
+                If enabled, it is recommended that
+                Style/UnlessElse and Style/EmptyElse be enabled.
+                This will conflict with Style/EmptyElse if
+                Style/EmptyElse is configured to style both
```
Comment (bbatsov) [inline config/disabled.yml]: You might quote `both` to make it stand out.

**Comment by rrosenblum** [comment]
@bbatsov the documentation has been updated and the code has been rebased.

**Comment by bbatsov** [comment]
:+1: Thanks!

**Pull request merged**